### ISSUE TITLE
[HOLD for payment 2025-01-15] [$250]  Notification displays the `@expensify.sms` domain in the title alongside the phone number.

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#53877]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.73-6
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @jayeshmangwani 
**Slack conversation** (hyperlinked to channel name): [expensify_bugs](https://expensify.slack.com/archives/C049HHMV9SM/p1733825991146889)

## Action Performed:
1. Log in with a phone number as User A.
2. Log in with any other account as User B.
3. Send a message from User A to User B.
4. On User B's device, the notification includes [`@expensify.sms](mailto:`@expensify.sms)` along with the phone number.

## Expected Result:
Notification should display only the sender's phone number.

## Actual Result:
Notification includes both the @expensify.sms domain and the sender's phone number.

## Workaround:
Unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [x] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

<img src="https://github.com/user-attachments/assets/6703f370-1049-4260-9cb9-cb0794562d23" width="40%" height="40%">

<img src="https://github.com/user-attachments/assets/81b82073-8643-4163-80ca-194a1f06e5f7" width="40%" height="40%">

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021866600416959560133</li>
        <li>Upwork Job ID: 1866600416959560133</li>
        <li>Last Price Increase: 2024-12-10</li>
<li>Automatic offers: </li>
<ul>
<li>truph01 | Contributor | 105376447</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sakluger</details>

**Comment by JoshIri360** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When a user with a phone number login sends a message, the notification displays their phone number with an unwanted "@expensify.sms" suffix (e.g., "+15857527441@expensify.sms"), making notifications look messy and unprofessional.

### What is the root cause of that problem?
The `createDisplayName` function in PersonalDetailsUtils wasn't removing the "@expensify.sms" domain before formatting the phone number for display in notifications. This domain is used internally for SMS-based logins but should never be shown to end users.

### What changes do you think we should make in order to solve the problem?
We modified the `createDisplayName` function in `PersonalDetailsUtils.ts`:

```typescript:src/libs/PersonalDetailsUtils.ts
function createDisplayName(login: string, passedPersonalDetails: Pick<PersonalDetails, 'firstName' | 'lastName'> | OnyxInputOrEntry<PersonalDetails>): string {
    // If we have a number like +15857527441@expensify.sms then let's remove @expensify.sms and format it
    const userLogin = LocalePhoneNumber.formatPhoneNumber(Str.removeSMSDomain(login));

    if (!passedPersonalDetails) {
        return userLogin;
    }
    // ... rest of function
}
```

This ensures that only the clean, formatted phone number appears in notifications and throughout the app.

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?
1. Test phone number formatting when user logs in with:
   - US phone number
   - International phone number
   - Phone number with "@expensify.sms" suffix
2. Test notification display when:
   - Receiving a message from phone number user
   - Viewing chat history with phone number user
3. Test edge cases:
   - Empty phone number
   - Phone number without "@expensify.sms" suffix
   - Multiple "@" symbols in the phone number

### What alternative solutions did you explore? (Optional)
N/A - The direct approach of cleaning the phone number at the display formatting level was the most straightforward solution.

**Comment by truph01** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
- Notification includes both the @expensify.sms domain and the sender's phone number.

### What is the root cause of that problem?
- The notification's title is from:

https://github.com/Expensify/App/blob/d9be98df93045f4a05419d60a822d5dee7c44c55/src/libs/Notification/LocalNotification/BrowserNotifications.ts#L102

and we didn't remove the sms domain in here.
### What changes do you think we should make in order to solve the problem?
- In [here](https://github.com/Expensify/App/blob/d9be98df93045f4a05419d60a822d5dee7c44c55/src/libs/Notification/LocalNotification/BrowserNotifications.ts#L102) we should remove the sms domain by using ```Str.removeSMSDomain(f.text)```.

- Similar should be applied in other platform

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


### What alternative solutions did you explore? (Optional)


- We can use ```LocalePhoneNumber.formatPhoneNumber(f.text)``` as well.

**Comment by truph01** [comment]
@parasharrajat Could you help check my [comment](https://github.com/Expensify/App/issues/53877#issuecomment-2538331396) above, please? Thanks

**Comment by sakluger** [comment]
Oh wow, I missed that you had already created and merged a PR, thanks @rlinoz!

**Comment by sakluger** [comment]
@parasharrajat do you know how we usually handle these cases where you reviewed proposals but didn't end up reviewing the PR? Are you due any payment here?

**Comment by sakluger** [comment]
Got it, thanks for confirming.

We can close as soon as the PR is deployed to prod.

**Comment by rlinoz** [comment]
PR is in staging

**Comment by rlinoz** [comment]
Hmm yeah, I think this should be testable in staging, but not 100% sure.

**Comment by sakluger** [comment]
@rlinoz can you also still reproduce this issue?

**Comment by rlinoz** [comment]
Yep I can, @truph01 is correct, I didn't realize the notification title for web is built in the FE, even though @truph01 said exactly that 😅 

The backend PR fixed a different issue in the end.

Assigning  you to the issue

**Comment by rlinoz** [comment]
Ah weird, it works for me in staging at least, might be something related to the number formatting maybe? I will try in dev with the same number you tested

![WhatsApp Image 2024-12-20 at 10 03 00](https://github.com/user-attachments/assets/0c026f5e-81fb-46da-8756-e28e034314a1)

**Comment by sakluger** [comment]
Hey @parasharrajat, could you please finish the BZ checklist? Ideally, it can be finished ahead of the payment due date so we don't need to wait to close the issue.

**Comment by sakluger** [comment]
Summarizing payment on this issue:

Contributor: @truph01 $250, paid via Upwork
Contributor+: @parasharrajat $250, please request on Newdot after completing the BZ checklist.

**Comment by sakluger** [comment]
Looks good, thanks!

**Comment by JmillsExpensify** [comment]
$250 approved for @parasharrajat
